docs/api-testing/composer/expressions.md

```diff
@@ -0,0 +1,234 @@
+
+Expressions are fields that reference an item in the test scope. The item can be a variable or an inner value in a data structure, such as a JSON.
+
+## What You'll Need
+* A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
+* Familiarity with the [API Testing Composer](/api-testing/composer/).
+
+## Using Expressions
+
+When working with structured data, expression is the path for reaching out a specific element. Most of the time, it's just object dot notation. Most expressions will start with the name of the variable the data is stored in. In this example, we will assume the data has been assigned to a variable named `payload`:
+
+```json
+"data":{
+  "created_time": "1453198835",
+  "images": {
+    "thumbnail": {
+      "width": 150,
+      "url": "https://domain.com/photos/9451802655724661601789699.jpg",
+      "height": 150
+      }
+   },
+  "Total-items": 1
+}
+```
+
+If you want to reach the value of the `created_time` attribute:
+
+```js
+payload.data.created_time
+```
+
+If you want to check the `width` of the images:
+
+```js
+payload.data.images.thumbnail.width
+```
+
+### Special Characters
+
+The `Total-items` element is a bit tricky, because the minus sign ( - ) would be misunderstood and treated as a subtraction operation. For this reason, the dot notation would require square brackets:
+
+```js
+data['Total-Items']
+```
+
+### XML
+
+The above statements are valid for both JSON and XML. When you have to reference XML attributes, the dot notation is valid, but the attribute has to be written between the square brackets and preceded by the `@`.
+
+Example:
+
+```xml
+<HotelList size="2">
+    <HotelSummary order="0">
+        <hotelId>20160502</hotelId>
+        <name>Hotel One</name>
+```
+
+If you want to check the `size` attribute, you have to write:
+
+```js
+payload['@size']
+```
+
+When you are working with SOAP API, the response might look something like this:
+
+```xml
+<?xml version="1.0"?>
+
+<soap:Envelope
+xmlns:soap="http://www.w3.org/2003/05/soap-envelope/"
+soap:encodingStyle="http://www.w3.org/2003/05/soap-encoding">
+
+<soap:Body>
+  <m:GetUserResponse>
+    <m:Username>Tony Stark</m:Username>
+  </m:GetUserResponse>
+</soap:Body>
+
+</soap:Envelope>
+```
+
+You'll also need to include the namespace in the expression. In the above scenario, if you want to check the `Username` item, you have to write:
+
+```js
+payload['soap:Body']['m:GetUserResponse']['m:Username']
+```
+
+:::note
+In XML, the root element is the variable itself, so you won't need to reference it explicitly.
+:::
+
+
+### Functions
+
+Expressions can also contain directives to transform the data you are willing to evaluate. For instance:
+
+```js
+<HotelList size="2">
+    <HotelSummary order="0">
+        <hotelId>20160502</hotelId>
+        <name>Hotel One</name>
+```
+
+```js
+payload.HotelSummary.size()
```

Comment:
@aliciakrebs replace `payload.HotelSummary.size()` with `payload.HotelSummary.length' (the new engine is javascript-based therefore length is the correct function)